src/test/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidatorTest.java

```diff
@@ -338,6 +338,24 @@ public void invalidCollection_notObject() throws Exception {
         assertThat(result1.getFieldId(), equalTo("Initials.1"));
```

Comment:
we need to introduce more tests. We have made an important change in the class logic. Please test it well

Reply:
DONE